src/backend/binaryen/lib/init_builtin_api.ts

```diff
@@ -80,6 +91,232 @@ function anyrefCond(module: binaryen.Module) {
     return module.block(null, [stmt], binaryen.i32);
 }
 
+function getPropNameThroughMeta(module: binaryen.Module) {
+    const objIndex = 0,
+        elemsIndex = 1,
+        metaIndex = 2,
+        metaFieldsCountIndex = 3,
+        metaFiledsPtrIndex = 4,
+        propNameIndex = 5,
+        fieldFlagIndex = 6,
+        loopIndex = 7,
+        iterPropCountIndex = 8,
+        curCharIndex = 9,
+        strLenIndex = 10;
+
+    const obj = module.local.get(objIndex, baseStructType.typeRef);
+    const elems = module.local.get(
+        elemsIndex,
+        stringArrayTypeInfoForStringRef.typeRef,
+    );
+    const meta = module.local.get(metaIndex, binaryen.i32);
+    const mataFieldsCount = module.local.get(
+        metaFieldsCountIndex,
+        binaryen.i32,
+    );
+    const metaFieldsPtr = module.local.get(metaFiledsPtrIndex, binaryen.i32);
+    const propName = module.local.get(propNameIndex, binaryen.i32);
+    const fieldFlag = module.local.get(fieldFlagIndex, binaryen.i32);
+    const loopIdx = module.local.get(loopIndex, binaryen.i32);
+    const iterPropCount = module.local.get(iterPropCountIndex, binaryen.i32);
+    const curChar = module.local.get(curCharIndex, binaryen.i32);
+    const strLen = module.local.get(strLenIndex, binaryen.i32);
+
+    const statementArray: binaryen.ExpressionRef[] = [];
+
+    // 1. get meta
+    const vtable = binaryenCAPI._BinaryenStructGet(
+        module.ptr,
+        StructFieldIndex.VTABLE_INDEX,
+        obj,
+        binaryen.i32,
+        false,
+    );
+    const metaValue = binaryenCAPI._BinaryenStructGet(
+        module.ptr,
+        VtableFieldIndex.META_INDEX,
+        vtable,
+        binaryen.i32,
+        false,
+    );
```

Comment:
I think "get vtable" and "get meta" should become two helper functions

Reply:
Good idea.